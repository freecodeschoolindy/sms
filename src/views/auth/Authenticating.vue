<template>
  <div>Authenticating...</div>
</template>

<script>
import { mapActions } from 'vuex';

export default {
  methods: {
    ...mapActions([
      'login'
    ])
  },
  mounted: function () {
    const authCode = this.$route.query.code;
    this.login({ authCode });
  }
};
</script>
