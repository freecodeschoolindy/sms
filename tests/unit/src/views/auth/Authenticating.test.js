import axios from 'axios';

import Authenticating from '@/views/auth/Authenticating.vue';

import testHelpers from '@@/testHelpers.js';
const urlMapper = testHelpers.urlMapper;

describe('Authenticating.vue', () => {
  let localVue;
  let store;
  let setup;

  let permissions = ['student'];

  beforeEach(() => {
    setup = testHelpers.setupTestVariables(permissions);
    localVue = setup.localVue;
    store = setup.store;

    axios.post = jest.fn((url) => {
      const urlMap = {
        '/auth/login/': { data: { access_token: '1234' } }
      };
      return urlMapper(url, urlMap);
    });

    axios.get = jest.fn((url) => {
      const urlMap = {
        'https://api.github.com/user?access_token=1234': { data: { id: 20, name: 'TEST' } }
      };
      return urlMapper(url, urlMap);
    });
  });

  test('Renders correctly', async () => {
    const wrapper = await testHelpers.mountWrapper(Authenticating, {
      localVue,
      store
    });

    expect(wrapper)
      .toMatchSnapshot();
  });
});
