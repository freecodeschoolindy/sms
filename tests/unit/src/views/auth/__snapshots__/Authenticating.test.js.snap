// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Authenticating.vue Renders correctly 1`] = `<div>Authenticating...</div>`;
